algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/luhn/LuhnChecker.java

```diff
@@ -0,0 +1,75 @@
+package com.baeldung.algorithms.luhn;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LuhnChecker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LuhnChecker.class);
+    
+    /*
+     * Starting from the rightmost digit, we add all the digits together, performing
+     * a special step for every second digit.
+     * 
+     * If the result is not divisible by 10, then the card number must not be valid.
+     * 
+     * We can form a number that passes the Luhn Check by subtracting the result of
+     * the Luhn algorithm from 10.
+     * 
+     * This is how the final digit of a credit card is calculated.
+     */
+    public static boolean checkLuhn(String cardNumber) {
+        int sum = 0;
+
+        try {
+
+            for (int i = cardNumber.length() - 1; i >= 0; i--) {
+
+                int digit = Integer.parseInt(cardNumber.substring(i, i + 1));
```

Comment:
I think we need to tighten up this spacing as well

```suggestion
        try {
            for (int i = cardNumber.length() - 1; i >= 0; i--) {
                int digit = Integer.parseInt(cardNumber.substring(i, i + 1));
```